public/wp-content/themes/ColoredCow/functions.php

```diff
@@ -56,40 +63,113 @@ function add_subscriber(){
 
 function verify_credentials(){
     global $wpdb;
+    if(isset($_POST['password'])&&isset($_POST['eventID'])):
     $tablename = $wpdb->prefix.'postmeta';
     $password = $_POST['password'];
-    $postID = $wpdb->get_var("SELECT post_id FROM $tablename WHERE meta_value = '".$_POST['guest_email']."'");
-    $status = get_post_meta($postID,'status',true);
-    $hash = get_post_meta($postID,'password',true);
-    if( wp_check_password( $password, $hash)){
-        if ($status=='confirmed') {
-            wp_send_json_error("duplicate");
-        }   
-        else{   
-            $wpdb->update(
-                $tablename,
-                array(
-                    'meta_value'=>'confirmed'
-                ),
-                array(
-                    'post_id'=>$postID,
-                    'meta_key'=>'status'
-                    )
-                ); 
-            wp_send_json_success("success");
-        }
-     }       
-    else{
-       wp_send_json_error("failed");
-    }    
+    $event_id = $_POST['eventID'];
+    $meta_key = 'event_users';
+    $event_users = get_post_meta( $event_id, $meta_key, true );
+    $rsvp_date = date("d/m/y");
+    $rsvp_guest_id = $wpdb->get_var("SELECT post_id FROM $tablename WHERE meta_value = '".$_POST['guest_email']."'");
+    $hash = get_post_meta($rsvp_guest_id,'password',true);
+
+    if(!wp_check_password( $password, $hash)){
+        wp_send_json_error("failed");
+    }
+    if(array_key_exists($rsvp_guest_id,$event_users)){
+        wp_send_json_error("duplicate");
+    }
+    $event_users = get_post_meta( $event_id, $meta_key, true );
+    $guest_name = get_the_title( $rsvp_guest_id );
+    $event_users[$rsvp_guest_id] = get_rsvp_guest_meta( $guest_name );
+    $event_users ? update_post_meta( $event_id, $meta_key, $event_users ) : add_post_meta($event_id, $meta_key, $event_users);
+    wp_send_json_success("success");
+    endif;
+    wp_die();
 } 
 add_action('wp_ajax_verify_credentials','verify_credentials'); 
 add_action('wp_ajax_nopriv_verify_credentials','verify_credentials');
 
-function check_duplicate_entry($phone,$email){
+function check_duplicate_entry($event,$phone,$email){
```

Comment:
Are we not using $event inside this function? If not, I don't think it's needed then.